scipy/special/specfun.c

```diff
@@ -0,0 +1,8417 @@
+                *zf = c;
+                *zd = ccos(0.5*pi*z*z);
+                return;
+            }
+            wa0 = wa;
+        }
+    } else if ((w0 > 2.5) && (w0 < 4.5)) {
+        m = 85;
+        c = z0;
+        cf1 = z0;
+        cf0 = 1.0e-100;
+        for (k = m; k >= 0; k--) {
+            cf = (2.0*k + 3.0)*cf0/zp - cf1;
+            if (k % 2 == 0) { c += cf; }
+            cf1 = cf0;
+            cf0 = cf;
+        }
+        c *= 2.0/(pi*z)*csin(zp)/cf;
+    } else {
+        // See comment at CFS(), use C(z) = iC(-iz)
+        if ((cimag(z) > -creal(z)) && (cimag(z) <= creal(z))) {
+            // right quadrant
+            d = 0.5;
+        } else if ((cimag(z) > creal(z)) && (cimag(z) >= -creal(z))) {
+            // upper quadrant
+            d = 0.5*I;
+        } else if ((cimag(z) < -creal(z)) && (cimag(z) >= creal(z))) {
+            // left quadrant
+            d = -0.5;
+        } else {
+            d = -0.5*I;
+        }
+        cr = 1.0;
+        cf = 1.0;
+        for (k = 1; k <= 20; k++) {
+            cr = -0.25*cr*(4.0*k - 1.0)*(4.0*k - 3.0)/zp2;
+            cf += cr;
+        }
+        cr = 1.0/(pi*z*z);
+        cg = cr;
+        for (k = 1; k <= 12; k++) {
+            cr = -0.25*cr*(4.0*k + 1.0)*(4.0*k - 1.0)/zp2;
+            cg += cr;
+        }
+        c = d + (cf*csin(zp) - cg*ccos(zp))/(pi*z);
+    }
+    *zf = c;
+    *zd = ccos(0.5*pi*z*z);
+    return;
+}
+
+
+void specfun_cfs(double complex z, double complex *zf, double complex *zd) {
+
+    // =========================================================
+    // Purpose: Compute complex Fresnel Integral S(z) and S'(z)
+    // Input :  z  --- Argument of S(z)
+    // Output:  ZF --- S(z)
+    //          ZD --- S'(z)
+    // =========================================================
+
+    int k, m;
+    double wb0, wb;
+    double complex s, cr, cf, cf0, cf1, cg, d;
+    const double eps = 1.0e-14;
+    const double pi = 3.141592653589793;
+
+    double w0 = cabs(z);
+    double complex zp = 0.5 * pi * z * z;
+    double complex zp2 = zp * zp;
+    double complex z0 = 0.0;
+
+    if (z == z0) {
+        s = z0;
+    } else if (w0 <= 2.5) {
+        s = z * zp / 3.0;
+        cr = s;
+        wb0 = 0.0;
+        for (k = 1; k <= 80; k++) {
+            cr = -0.5 * cr * (4.0 * k - 1.0) / k / (2.0 * k + 1.0) / (4.0 * k + 3.0) * zp2;
+            s += cr;
+            wb = cabs(s);
+            if ((fabs(wb - wb0) < eps) && (k > 10)) {
+                    *zf = s;
+                    *zd = csin(0.5*pi*z*z);
+                    return;
+            }
+            wb0 = wb;
+        }
+    } else if ((w0 > 2.5) && (w0 < 4.5)) {
+        m = 85;
+        s = z0;
+        cf1 = z0;
+        cf0 = 1.0e-100;
+        for (k = m; k >= 0; k--) {
+            cf = (2.0*k + 3.0)*cf0/zp - cf1;
+            if (k % 2 == 1) { s += cf; }
+            cf1 = cf0;
+            cf0 = cf;
+        }
+        s = 2.0*s/(pi * z)*csin(zp)/cf;
```

Comment:
```suggestion
        s = 2.0/(pi * z)*csin(zp)/cf*s;
```

https://github.com/scipy/scipy/blob/829d0dd950d63caece04526b3a617bbe2bc24023/scipy/special/specfun/specfun.f#L133